File: Solve the Sudoku.cpp
 public:
bool valid(int grid[N][N], int row, int col, int c) {
        for(int i=0;i<N;i++) {
            if(grid[row][i] == c) {
                return false;
            }
            if(grid[i][col]==c) {
                return false;
            }
            if(grid[3*(row/3)+i/3][3*(col/3)+i%3]==c) {
                return false;
            }
        }
        return true;
    }
    bool SolveSudoku(int grid[N][N])  
    { 
        for(int i=0;i<N;i++) {
            for(int j=0;j<N;j++) {
                if(grid[i][j]== 0) {
                    for(int c=1;c<=9;c++) {
                        if(valid(grid,i,j,c)) {
                            grid[i][j]=c;
                            
                           if(SolveSudoku(grid)==true) {
                               return true;
                           }
                           else {
                               grid[i][j] = 0;
                           }
                        }
                    }
                    return false;
                }
            }
        }
        return true;
    }
    
    //Function to print grids of the Sudoku.
    void printGrid (int grid[N][N]) 
    {
       SolveSudoku(grid);
       for(int i=0;i<N;i++) {
           for(int j=0;j<N;j++) {
               cout << grid[i][j] << " ";
           }
           
       }
    }

File: Solve the Sudoku.java
class Solution
{
    static boolean isSafe(int row , int col ,int  grid[][], int val )
    {
        int n = grid.length;
        for (int i =0; i<n ;i++){
            if (grid[row][i]==val)
            return false;
            if (grid[i][col]== val)
            return false;
            // 3*3 check martix
            if (grid[3*(row/3) +i/3][3*(col/3)+i%3]== val)
            return false ;
        }
        return true;
    }
    //Function to find a solved Sudoku. 
    static boolean SolveSudoku(int grid[][])
    {
        int n = grid.length;
        for (int row =0; row <n; row++){
            for (int col =0; col<n; col++){
                if (grid[row][col]==0){
                    for (int val =1; val <=9 ; val++){
                        if (isSafe(row, col , grid, val)){
                            grid[row][col] = val;
                            
                        boolean pendingSolution = SolveSudoku(grid);
                        if (pendingSolution)
                        return true;
                        else {
                           grid[row][col] =0; 
                        }
                        
                        }
                    }
                    return false;
                }
            }
        }
        return true;
    }
    
    //Function to print grids of the Sudoku.
    static void printGrid (int grid[][])
    {
        // add your code here
        SolveSudoku(grid);
        if (SolveSudoku(grid)) {
        for (int row = 0; row < grid.length; row++) {
            for (int col = 0; col < grid.length; col++) {
                System.out.print(grid[row][col]+" ");
            }
            
        }
    } else {
        System.out.println("No solution exists.");
    }
        
    }
}
